### DISCUSSION TITLE
Idea: teach the thing to say it does not know, and the embarrassing question underneath

### DISCUSSION THREAD

**Discussion by github-actions** [fde-academy-lab/advanced-rag-lab#138] [Ideas]
> 💬 **Dan** · <sub>career-switcher · 18 months in</sub>

Asking the naive version first because I genuinely do not know the answer.

The corpus has 243 questions. 207 are answerable and 36 are unanswerable by construction. The
baseline scorecard says:

```
abstention_recall        0.0
abstention_f1            0.0
false_answers_on_null    36
over_refusals             0
```

We answer all 36 of them. Confidently, with citations, in the same format as a correct answer.

**The embarrassing question.** Why is `over_refusals 0` printed next to that as if it were good
news? A system that never refuses gets a perfect over-refusal score for the same reason a system
that never predicts positive gets perfect precision on the empty set.

**The idea**, extension 10 on the research page: self-critique. Generate, then have the model
judge whether the retrieved evidence supports what it just wrote, and re-retrieve or abstain if
it does not.

**Metric and slice.** `abstention_f1` over the 36 nulls, with `over_refusals` on the 207
answerable reported beside it every time. Neither on its own means anything.

**Effect that would count.** The page sets the bar at abstention F1 above 0.38 and calls this
the highest-value open item in the repo.

**Cost.** An extra model call per query. Mean `tokens_in` is 884.05 with p95 1123, so a critique
pass over the same context roughly doubles the input side of the bill, and adds a component that
can regress on its own.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 💬 **Lena** · <sub>research background · reads the papers</sub>

Self-RAG, Asai et al. 2023, is exactly this and it is further along
than you think. The model emits reflection tokens during generation that say whether retrieval
was needed, whether the passage supports the claim, and whether the output is useful. You get
the critique inline rather than as a second pass, so the cost story is better than doubling.

The reported gains on their evaluation are large and they hold on the open-domain sets. This is
one of the cleanest results in the retrieval literature over the last few years.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 💬 **Dan** · <sub>career-switcher · 18 months in</sub>

That settles it then, and it is more tractable than I expected if the
critique comes inline. Scoping it as a two-week piece:

1. wire the reflection tokens into the generate step
2. threshold on the support token
3. abstain below the threshold
4. re-run the baseline and report `abstention_f1` against the 0.38 bar

Week one is 1 and 2, week two is 3 and the re-run. If it lands above 0.38 we ship it and close
EX-18, and if it lands under we have a threshold to tune rather than a design to rethink, which
is the good kind of not-yet. Anyone want to review the wiring before I start?

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 💬 **Marcus** · <sub>data scientist · strong statistics</sub>

Dan, the measurement you would put between step 0 and step 1 has
already been run, and it deletes most of the plan rather than confirming it.

**No signal we compute separates the 36 nulls from the 207 answerable.** Best F1 **0.38** across
four of them — top-1 score, the rank-1-to-rank-2 gap, the mean of the top-k, and score entropy —
and the metrics page says so with the mechanism attached: the nulls *name real entities in the
corpus's own vocabulary* while the genuine questions paraphrase, so the unanswerable ones come
out lexically **closer** to the corpus. Any threshold on retrieval score is reading a feature
with the wrong sign, and no amount of tuning repairs a sign.

That matters for your step 1, because a critique layer has to learn the separation from
something. If the retrieval signal carries none — and it does not — a second model call over the
same eight chunks is looking at the same undifferentiated evidence the first one saw. You would
be paying twice for one view.

Which leaves the question the repository actually has open: whether a **sufficiency** signal —
does this evidence *entail* an answer — carries what a similarity signal cannot. That is issue
#10 and nobody has measured it. It is the only part of this idea that is genuinely unknown, and
it is worth more than the four steps around it.

Second point, on reporting. 36 items is a small denominator, and no interval has ever been put
on that 0.38 — the repository publishes the point estimate and nothing around it. Decide now how
you will report it, and pre-commit to the interval rather than the point.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 💬 **Tomás** · <sub>SRE</sub>

And the direction nobody has said out loud. `over_refusals` is 0
today, which means every abstention you add is a brand new way for this thing to fail in front
of a user, and it is a failure mode the current system is structurally incapable of.

I am not arguing against it. I am arguing that the gate has to move at the same time. Gate on
both numbers or you will trade 36 confident wrong answers for some unknown number of "I could not
find that" against evidence that was sitting in the context.

Also make the abstention message name what was missing. A refusal that says nothing is a support
ticket; a refusal that says which entity or date it could not find is a search the user can
redo.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 🛠 **Maintainer**

Lena's summary of the paper is accurate and the setup is the
part that does not carry over.

Self-RAG's reflection tokens come from a model **trained to emit them**, on data built for that
purpose. The training set is the contribution. The inference-time loop is the cheap half, and
bolting the loop onto a model that was never trained for it gives you a second generation pass
that shares the first pass's priors, including the prior that produced the confident answer you
are asking it to doubt. A critic with the same beliefs as the author agrees with the author.

The condition under which the published result transfers is that you have supervision for the
critique step, or can afford to make it. That is the same conclusion Marcus arrives at from the
other side: his separation check is asking whether the supervision already exists for free in
the scores.

One caution on the target, and it points the opposite way to the obvious one. These 36 nulls are
not the easy end of the abstention problem — they are adversarial by construction. They are
on-topic, in-vocabulary, and they retrieve confident well-ranked evidence, which is exactly why
similarity-based abstention cannot touch them. Real unanswerable questions are ones where plausible evidence
exists and does not support the claim. An abstention F1 measured only on constructed nulls will
read higher than the same system scores in production, so treat 0.38 as a floor on a friendly
set rather than as the finish line.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>

**Comment by github-actions** [comment]
> 💬 **Dan** · <sub>career-switcher · 18 months in</sub>

Right, and my two-week plan had no measurement in it.

Four build steps, one re-run at the end, and the re-run was there to confirm the thing rather
than to decide anything. I accepted the first confident answer in the thread and started
scheduling. Lena's answer was correct about a different question from the one I asked, and I could not tell
the difference because I did not know enough to ask which setup it came from.

Rewritten:

- **Day one:** read the result that already exists rather than reproducing it — best F1 0.38,
  and the nulls are lexically *closer* to the corpus than the answerable questions. The branch I
  had written as "if they separate" is closed, and I was going to spend a day closing it again.
- **Day two:** build the sufficiency check as a **standalone component** — does this evidence
  entail an answer — and report its precision and recall against the 36 before wiring it into
  anything. That is EX-18's acceptance criteria and it is the first genuinely unmeasured step.
- **Only then:** the critique layer, and only if the standalone number justifies it.
- **Either way:** `abstention_f1` and `over_refusals` reported together, on 36 and 207
  respectively, and I will say plainly that no interval has been put on either.

Posting the standalone numbers whichever way they come out.

---
<sub>📎 **Worked example.** Written by the maintainers to model what a good question, a good wrong turn and a good correction look like. Not a real cohort member — see the personas register in docs/10-community/. Start your own thread rather than replying here unless you have something to add.</sub>
